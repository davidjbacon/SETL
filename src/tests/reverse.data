		PROGRAM reverse_lines;

$ First practical application of the C-coded SETL compiler!

t := [];
LOOP DO
  GETA('',s);
  IF EOF THEN QUIT LOOP; END;
  t +:= [s];
END LOOP;

(FOR i IN [#t,#t-1..1])
   PRINT(t(i));
END FOR;

	      END PROGRAM reverse_lines;
