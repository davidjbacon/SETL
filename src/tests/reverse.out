	      END PROGRAM reverse_lines;

END FOR;
   PRINT(t(i));
(FOR i IN [#t,#t-1..1])

END LOOP;
  t +:= [s];
  IF EOF THEN QUIT LOOP; END;
  GETA('',s);
LOOP DO
t := [];

$ First practical application of the C-coded SETL compiler!

		PROGRAM reverse_lines;
